### ISSUE TITLE
[Bug]: Calendar Notification for new Event: Link not working

### ISSUE THREAD

**Issue by bentuna** [nextcloud/server#43430]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When you get a notification for a new event, someone else created in a shared calendar, and you click on the link to the event in the notifications overlay or in the activities app, the calendar app opens, but the sidebar shows "Event does not exist It might have been deleted, or there was a typo in a link".

### Steps to reproduce

1. Person A: Enable Push notifications for "A calendar event was modified" in your personal settings
2. Person B: Share a calendar with Persan A
3. Person B: Create an event inside the shared calendar
4. Person A: notice the notification "B created event X in calendar Y" and click on the Link "X ↗"

### Expected behavior

The link should open the event details of the linked event in the calendar sidebar.

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

28

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by ChristophWurst** [comment]
do you have the event in your calendar? is the calendar enabled?

**Comment by ChristophWurst** [comment]
Can you compare the URLs of manually selecting vs the one from activity? Something must be wrong/different.

**Comment by bentuna** [comment]
> Can you compare the URLs of manually selecting vs the one from activity? Something must be wrong/different.

Yes, the link from **notifications** and **activity app** goes to:

`https://portal.101lab.it/apps/calendar/dayGridMonth/now/edit/sidebar/L3JlbW90ZS5waHAvZGF2L2NhbGVuZGFycy9oYWxsb0Bvcmdpcy5iZS9iZW4tdGVzdC9ENDkxQkQ4RS0zNEMxLTQwMTctODIyNS1GQzFBOEM1RjE3NUQuaWNz/next`

The URL when opening the calendar event **manually** is:

`https://portal.101lab.it/apps/calendar/dayGridMonth/now/edit/sidebar/L3JlbW90ZS5waHAvZGF2L2NhbGVuZGFycy9iZW5qYW1pbi9iZW4tdGVzdF9zaGFyZWRfYnlfaGFsbG9Ab3JnaXMuYmUvRDQ5MUJEOEUtMzRDMS00MDE3LTgyMjUtRkMxQThDNUYxNzVELmljcw==/1708387200`

**Comment by nextcloud-command** [comment]
This issue has been automatically marked as stale because it has not had recent activity and seems to be missing some essential information. It will be closed if no further activity occurs. Thank you for your contributions.

**Comment by bentuna** [comment]
@ChristophWurst - it would be nice to reopen, because the problem still persists in Nextcloud 28.0.3 with Calendar 4.6.7.

**Comment by nkinkade** [comment]
For what its worth, I have the same issue with Nextcloud 28.0.4, and Calendar 4.6.7. Calendar links have never worked for me in email notifications for a shared calendar, even with many previous versions of Nextcloud and Calendar. This doesn't appear to be a regression, but something that may have never worked, at least with my setup, which is very vanilla. The links in email notifications, as well as the links from the Activity page are both the same, and both don't work for any given event.

**Comment by nkinkade** [comment]
As an example, and very similar to @bentuna, the link in the notification email is something like:

```
https://cloud.example.com/index.php/apps/calendar/dayGridMonth/now/edit/sidebar/L3JlbW90ZS5waHAvZGF2L2NhbGVuZGFycy9rYXRobGVlbi9wZXJzb25hbC8zRDdFQTI3NS1GRkYyLTRFRTItQUZERS02OUIxMEYyOTc0Q0UuaWNz/next
```

While the URL of the actual calendar event it is supposed to be pointing toward is:

```
https://cloud.example.com/index.php/apps/calendar/dayGridMonth/now/edit/sidebar/L3JlbW90ZS5waHAvZGF2L2NhbGVuZGFycy9ua2lua2FkZS9wZXJzb25hbF9zaGFyZWRfYnlfa2F0aGxlZW4vM0Q3RUEyNzUtRkZGMi00RUUyLUFGREUtNjlCMTBGMjk3NENFLmljcw==/1713544200
```

I notice that the "objectId" of the URLs both contain the prefix `L3JlbW90ZS5waHAvZGF2L2NhbGVuZGFycy9`, but differ after that, with the working URL's objectId being a good deal longer. Maybe that prefix denotes the calendar? I tried to take a peek at the code, but having zero familiarity with how the calendar app works, nor notifications, I decided that it would probably be far more expedient for someone who knows how those "objectId"s are created and what they represent to look into the issue. That said, I am happy to help work on this in any way I can. If someone can give me some quick pointers that will bootstrap my understand of the workflow, then I may be able to help delve further into it. I wasn't able to find any sort of identifier even remotely similar to the URL "objectId" in the database entry for the email in question in the oc_calendarobjects table.

**Comment by ChristophWurst** [comment]
> I notice that the "objectId" of the URLs both contain the prefix `L3JlbW90ZS5waHAvZGF2L2NhbGVuZGFycy9`, but differ after that, with the working URL's objectId being a good deal longer.

personal vs personal_shared_by_sharer

Former is the URL for the owner of the calendar, latter is the URL for those who the calendar is shared with

**Comment by nkinkade** [comment]
Thank you for the clarification. It sounds like the nature of the problem is well understood.

**Comment by joshtrichards** [comment]
Duplicate of nextcloud/calendar#5293
Fixed in #45775
